src/Support/Model.php

```diff
@@ -212,6 +224,21 @@ public function __unset($key)
         $this->offsetUnset($key);
     }
 
+    /**
+     * Return a timestamp as DateTime object.
+     *
+     * @param  mixed  $value
+     * @return \Illuminate\Support\Carbon
+     */
+    protected function asDateTime($value)
+    {
+        if (is_numeric($value) && $this->timestampsInMilliseconds) {
+            return Date::createFromTimestampMs($value);
```

Comment:
@jimmy4o4 Why is this `Date::createFromTimestampMs` & not `Carbon::createFromTimestampMs`?

I have never looked into fractional seconds with Laravel's timestamps, but from what I am seeing there are issues with a version of the PDO driver that can cause issues.

Do you know how backwards compatible this is?

Reply:
'Date' is the same facade as used in original asDateTime method from HasAttributes trait (Illuminate\Support\Facades\Date). Eventually the same Carbon class object is returned by this method as in other 'Date' facade methods. So it shoudn't break compatibility, just Carbon object now has information about milliseconds that ClickUp API returns.

More important is that the returned Carbon object now has the correct date, if someone stored old date somewhere it will change for him now.

I've been trying to use ClickUp API for only couple of weeks now and have no idea if timestamps in millisecond just appeared there, or were there from the very begining. API documentation is't clear about used data types and older API versions either.